Source/player.h

```diff
@@ -407,25 +407,33 @@ struct PlayerStruct {
 
 	/**
 	 * @brief Return player's melee to hit value
+	 * @param pierceArmor - indicate if player's armor piercing should be included - true vs monsters, false vs players
 	 */
-	int GetMeleeToHit() const
+	int GetMeleeToHit(bool pierceArmor = true) const
 	{
 		int hper = _pLevel + _pDexterity / 2 + _pIBonusToHit + BaseHitChance;
 		if (_pClass == HeroClass::Warrior)
 			hper += 20;
+		//in hellfire armor piercing ignores % of enemy armor instead, no way to include it here
+		if (!gbIsHellfire && pierceArmor)
+			hper += _pIEnAc;
 		return hper;
 	}
 
 	/**
 	 * @brief Return player's ranged to hit value
+	 * @param pierceArmor - indicate if player's armor piercing should be included - true vs monsters, false vs players
 	 */
-	int GetRangedToHit() const
+	int GetRangedToHit(bool pierceArmor = true) const

```

Comment:
Instead of bool could we make a variant:
```cpp
int GetRangedPiercingToHit()
{
	int hper = GetRangedToHit();
	if (!gbIsHellfire)
		hper += _pIEnAc;
	return hper;
}
```